rule bwa_map:
	input:
		genome_fa = "/biarchive/project/jhshin/reference/hg38/ucsc/hg38_chr1.fa", 
		fastq_1 = "/biarchive/project/jhshin/test/snakemake_tutorial/test_data/test_1.fastq.gz",
		fastq_2 = "/biarchive/project/jhshin/test/snakemake_tutorial/test_data/test_2.fastq.gz"
	output:
		bam = "/biarchive/project/jhshin/test/snakemake_tutorial/test_data/test.bam"
	params: 
		thread = 8
	shell:
		"""
		bwa mem -t {params.thread} {input.genome_fa} {input.fastq_1} {input.fastq_2} \
		| samtools view -Sb - \
		> {output.bam}
		"""
